taqueria-plugin-tezos-client/common.ts

```diff
@@ -0,0 +1,41 @@
+import { getArch, getLatestFlextesaImage } from '@taqueria/node-sdk';
+import { RequestArgs } from '@taqueria/node-sdk/types';
+import { join } from 'path';
+
+export interface TypeCheckOpts extends RequestArgs.ProxyRequestArgs {
+	sourceFile: string;
+}
+
+export interface SimulateOpts extends RequestArgs.ProxyRequestArgs {
+	sourceFile: string;
+	storage?: string;
+	param: string;
+	entrypoint?: string;
+}
+
+// To be used for the main entrypoint of the plugin
+export type IntersectionOpts = TypeCheckOpts & SimulateOpts;
+
+// To be used for common functions in this file
+type UnionOpts = TypeCheckOpts | SimulateOpts;
+
+const ENDPOINT = process.env['TAQ_TEZOS_CLIENT_RPC'] ?? 'https://ghostnet.ecadinfra.com';
+export const GLOBAL_OPTIONS = `--endpoint ${ENDPOINT}`;
+
+export const trimTezosClientMenuIfPresent = (msg: string): string => {
+	return msg.replace(/Usage:(.|\n)+/, '');
+};
+
+export const getInputFilename = (opts: UnionOpts, sourceFile: string) =>
+	join('/project', opts.config.artifactsDir, sourceFile);
+
+export const getCheckFileExistenceCommand = async (parsedArgs: UnionOpts, sourceFile: string): Promise<string> => {
+	const projectDir = process.env.PROJECT_DIR ?? parsedArgs.projectDir;
```

Comment:
I don't think we should have plugins look for environment variables for core configuration settings. We should have the CLI do that so that the plugin can just use parsedArgs.

Reply:
What's wrong with that? A plugin author may do such a thing and it wouldn't be practical for them to wait for us to update the CLI to accommodate this no?

Reply:
For first-party products, we have some more control and options about where we do this kind of thing.

I think it would make sense to have the CLI interpret PROJECT_DIR rather than have each plugin potentially behave differently. If PROJECT_DIR environment variable is absolutely needed here, then I think it belongs in the CLI. Otherwise, I'd just remove checking for that environment variable now and we can add a separate task/PR for that effort. What do you think?

Reply:
Sounds good!